### DISCUSSION TITLE
Compare Script Kit and Alfred

### DISCUSSION THREAD

**Discussion by tung-dang** [johnlindquist/kit#722] [General]
I am not sure the Github discussion is a right way to share my personal thought or not. Feel free to close this thread and let me know where I could share my personal thought. Thank you!

Recently, I’ve experimented with Script Kit app. Script Kit app allows writing custom scripts in JS or TS to automate your daily task in macOS. The script is quite similar to custom workflows in [Alfred app](https://www.alfredapp.com/). In my personal opinion, as a FE dev, I feel it’s easier for me to write a script/workflow than writing in the Alfred app. Alfred app supports many languages to write workflow but I feel Alfred app is quite complex to write workflow for it.

Script Kit is still new so Script Kit still has some cons

- Performance - Slower than Alfred app. I hope the performance will improve in future.
- Need better documentation. Currently, I read its API via https://github.com/johnlindquist/kit/discussions.

While there are some pros: 

- Easy to write script/workflow
- Easy to debug script
- Focus only JS/TS and support executing Apple Script in JS/TS
- Free and open-source so I think there will be a lot of community script support in future.

I am trying to rewrite some of my Alfred workflows to Script Kit. I am using both Alfred and Script Kit. 
In summary, I really love the easy way to write scripts in Script Kit.

**Comment by johnlindquist** [comment]
Thanks for sharing, I'm always happy to hear personal thoughts about Script Kit!

I would love to hear more about your two cons:

1. Performance - can you please point to a specific scenario where you see Script Kit behaving slowly? It would really help me to know which areas to users are experiencing any issues like that
2. Better docs - Which specific things are you looking for documentation on? Is there a specific workflow/script I can help with?

I'd be very happy to teach how to convert workflows from any other tools to Script Kit. If you ever have any ideas or specific links you'd like to see in Script Kit, please share and I'll do what I can!

**Reply by tung-dang** [reply]
Thanks for your reply, @johnlindquist ! 

1. Performance. This is my personal feeling in general when I use both apps daily. For a simple task - opening a folder in VSCode. The task is `edit(folderPath)`. I feel the time to wait for opening VSCode in Alfred is faster than in Script Kit. 

Alfred supports to remember the last action. For example, I have Task X -> choose an action of Task X --> Choose a sub-action of Task X. At the first time, I chose an action in `Sub Task X`, if I trigger Alfred in a short time (maybe < 5 minutes) after that, Alfred will stand at the last task `Sub Task X` so I can choose another action in `Sub Task X`. With Script Kit, I don't know how to do the same. I always type Task X -> Choose an action of Task X --> Choose another `sub-action` of Task X again. 

2. Better docs. For example, I wanted to look for all global functions/variables I could use and read documents of each one. I usually search or go around Github Discussion to find. Videos you made in EggHead are super helpful to get started. Thanks a lot for that!

**Reply by johnlindquist** [reply]
Awesome, thanks for following up:

1. VS Code opening slowly:

The is usually an issue with a slow PATH lookup for `code`.

In `~/.kenv/.env`, trying switching from

```
KIT_EDITOR=code
```

to the full path

```
KIT_EDITOR=/usr/local/bin/code
```

Often tools like `nvm` make looking up `code` or other tools on the PATH much, MUCH slower. In your scripts, it's best to use the full path to the any command-line tool as well (if you're ever using `exec` or `$`.

Please let me know if that helps!

2. Docs - that's a great idea. I'll put together a "megadoc" with all the apis that Kit provides with the simplest possible examples.

**Reply by tung-dang** [reply]
It's a nice tip! `KIT_EDITOR=<full-path-to-code` helps to improve opening VSCode time. Thanks @johnlindquist !

**Reply by johnlindquist** [reply]
Awesome, glad it helped. The full path has been the default for the past couple releases, so new users shouldn't have to worry about it.

Any other feedback I can help with?
